src/components/numberSelector/NumberSelector.tsx

```diff
@@ -19,90 +21,122 @@ export interface Props {
+    decrementStep?: number;
 }
 
 export const NumberSelector = ({
+    validator,
+    initialValue,
     value,
-    onValueChange= undefined,
-    onEndEditing = undefined,
-    minValue = 0,
-    maxValue,
+    onEndEditing,
     style,
     minusIcon = 'minus',
     plusIcon = 'add',
     showSoftInputOnFocus = false,
     variant = 'outlined',
-    decimal = false,
     size = 'm',
-    placeholder = '--',
+    incrementStep = 1,
+    decrementStep = 1,
 }: Props) => {
+    const [tempValue, setTempValue] = useState<string>(value?.toString() ?? '');
+
+    const computeCrementedValue = (step: number): number => {
+        return roundValue(
+            Math.max(
+                Math.min(
+                    Number(getParsedValue() ?? 0) + step,
+                    validator.maxValue,
+                ),
+                validator.minValue,
+            ),
+        );
+    };
+
+    const computeIncrementedValue = (): number => {
+        return computeCrementedValue(incrementStep);
+    };
+
+    const computeDecrementedValue = (): number => {
+        return computeCrementedValue(-decrementStep);
+    };
+
+    const roundValue = (val: number): number => {
+        return Math.round(val * 10) / 10;
+    };
+
+    const getParsedValue = (): number => {
+        return validator.parser(tempValue);
+    };
+
+    const getValidValue = (): string => {
+        if (tempValue === '') {
+            return lastValidValue?.toString() ?? '';
+        }
+        return getParsedValue().toString();
+    };
+
     const refInput = useRef<any>();
-    const parser = decimal ? parseFloat : parseInt;
-    const [tempValue, setTempValue] = useState<string>(value?.toString() ?? placeholder);
+    const minusDisabled = validator.minValue === getParsedValue();
+    const addDisabled = validator.maxValue === getParsedValue();
+
+    const [lastValidValue, setLastValidValue] = useState<number>();
     const [softInputOnFocus, setSoftInputOnFocus] = useState(false);
-    const allowedMinus = (): boolean => {
-        return minValue !== undefined && minValue < 0;
-    };
-    const decimalRegex = allowedMinus() ? /^-?\d+[\.]?\d?$/ : /^\d+[\.]?\d?$/;
-    const integerRegex = allowedMinus() ? /^-?\d+$/ : /^\d+$/;
-    const numberRegex = decimal ? decimalRegex : integerRegex;
+
+    const [error, setError] = useState(false);
+    const [focused, setFocused] = useState(false);
+    const [fieldState, setFieldState] =
+        useState<FieldState>('filledWithDefault');
+
     const onAdd = () => {
         Keyboard.dismiss();
-        if (!addDisabled) {
-            const newValue =value?  (
-                Math.round(
-                    (value + 1 < maxValue ? value + 1 : maxValue) * 10
-                ) / 10
-            ).toString() : "1";
-            onChangeText(newValue);
-            onEndEditing && onEndEditing(newValue);
-        }
+        if (addDisabled) return;
+        const newValue = computeIncrementedValue().toString();
```

Comment:
on a un cas particulier qui n'est pas gérer : 

GIVEN le NumberSelector est en float
AND la valeur est a 3.2
WHEN l'utilisateur appuis sur '+'
THEN la valeur passe a 4

GIVEN le NumberSelector est en float
AND la valeur est a 3.2
WHEN l'utilisateur appuis sur '-'
THEN la valeur passe a 3